internal/pkg/handler/updated-handler.go

```diff
@@ -20,11 +32,210 @@ func (r ResourceUpdatedHandler) Handle() error {
 		// process resource based on its type
 		if _, ok := r.Resource.(*v1.ConfigMap); ok {
 			logrus.Infof("Performing 'Updated' action for resource of type 'configmap'")
+			rollingUpgrade(r, "configmaps", "deployments")
+			rollingUpgrade(r, "configmaps", "daemonsets")
+			rollingUpgrade(r, "configmaps", "statefulSets")
 		} else if _, ok := r.Resource.(*v1.Secret); ok {
 			logrus.Infof("Performing 'Updated' action for resource of type 'secret'")
+			rollingUpgrade(r, "secrets", "deployments")
+			rollingUpgrade(r, "secrets", "daemonsets")
+			rollingUpgrade(r, "secrets", "statefulSets")
 		} else {
-			logrus.Warnf("Invalid resource: Resource should be 'Secret' or 'Configmap' but found %v", r.Resource)
+			logrus.Warnf("Invalid resource: Resource should be 'Secret' or 'Configmap' but found, %v", r.Resource)
+		}
+	}
+	return nil
+}
+
+func rollingUpgrade(r ResourceUpdatedHandler, resourceType string, rollingUpgradeType string) {
+	client, err := kube.GetClient()
+	if err != nil {
+		logrus.Fatalf("Unable to create Kubernetes client error = %v", err)
+	}
+	var namespace, name, sshData, envName string
+	if resourceType == "configmaps" {
+		namespace = r.Resource.(*v1.ConfigMap).Namespace
+		name = r.Resource.(*v1.ConfigMap).Name
+		sshData = helper.ConvertConfigmapToSHA(r.Resource.(*v1.ConfigMap))
+		envName = "_CONFIGMAP"
```

Comment:
You don't need to write constants again and again for the same resource. Since you know the resource type, better use that to generate the env name